lib/declare_schema/model/index_definition.rb

```diff
@@ -69,12 +71,30 @@ def default_index_name(table, fields)
 
         private
 
+        SHA_SUFFIX_LENGTH = 4
+
+        def shorten_name(name, max_len)
+          if name.size <= max_len
+            name
+          else
+            name_prefix = name.first(max_len >= SHA_SUFFIX_LENGTH*2 ? (max_len - SHA_SUFFIX_LENGTH) : ((max_len + 1)/2))
+            sha = Digest::SHA256.hexdigest(name)
+            (name_prefix + sha).first(max_len)
```

Comment:
Tricky to understand what this will look like just by inspecting the code. Really appreciate the thorough test 💯

Reply:
Also, what would you think about a warning message if we have to truncate the unique name? It might be nice to point people towards the `name` option of the index dsl.

Reply:
Good point. I was just thinking that too! https://github.com/Invoca/declare_schema/pull/158/commits/9592b0ed42fc11733dcf4e0effe60b203ffab46d